python/cm/issue.py

```diff
@@ -10,18 +10,65 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from cm.logger import log   # pylint: disable=unused-import
+from collections import defaultdict
+
 import cm.status_api
-from cm.errors import AdcmEx
-from cm.errors import raise_AdcmEx as err
 from cm.adcm_config import proto_ref, obj_ref, get_prototype_config
-from cm.models import ConfigLog, Host, ClusterObject, Prototype, HostComponent
-from cm.models import PrototypeImport, ClusterBind
+from cm.errors import AdcmEx, raise_AdcmEx as err
+from cm.logger import log
+from cm.models import (
+    ClusterBind,
+    ClusterObject,
+    ConfigLog,
+    Host,
+    HostComponent,
+    Prototype,
+    PrototypeImport,
+    ServiceComponent,
+)
+
+
+def update_hierarchy_issues(obj):
+    save_issue(obj)
+
+    if not obj or obj.prototype.type == 'adcm':
```

Comment:
This check is repeated in the `save_issue()` function (line 66-70). I think you can remove them from the `save_issue()` function, and move the check call to line 32.